scarlet/morphology.py

```diff
@@ -110,51 +130,36 @@ def get_model(self, *parameters):
 
     def update(self):
         image = self._parameters[0]
-        size = max(image.shape)
 
         if not self.resizing or image.fixed:
             return
 
-        # shrink the box? peel the onion
-        dist = 0
-        while (
-            np.all(image[dist, :] == 0)
-            and np.all(image[-dist, :] == 0)
-            and np.all(image[:, dist] == 0)
-            and np.all(image[:, -dist] == 0)
-        ):
-            dist += 1
-
-        newsize = initialization.get_minimal_boxsize(size - 2 * dist)
-        if newsize < size:
-            dist = (size - newsize) // 2
-            # Create new parameter for smaller image
+        # shrink the box?
+        bbox = self.bbox.copy()
+        self.shrink_box(image)
+        if bbox != self.bbox:
+            slice, _ = overlapped_slices(bbox, self.bbox)
             image = Parameter(
-                image[dist:-dist, dist:-dist],
+                image[slice],
                 name=image.name,
                 prior=image.prior,
                 constraint=image.constraint,
                 step=image.step / 2,
                 fixed=image.fixed,
-                m=image.m[dist:-dist, dist:-dist] if image.m is not None else None,
-                v=image.v[dist:-dist, dist:-dist] if image.v is not None else None,
-                vhat=image.vhat[dist:-dist, dist:-dist]
-                if image.vhat is not None
-                else None,
+                m=image.m[slice] if image.m is not None else None,
+                v=image.v[slice] if image.v is not None else None,
+                vhat=image.vhat[slice] if image.vhat is not None else None,
             )
 
             # set new parameters
             self._parameters = (image,) + self._parameters[1:]
 
-            # adjust bbox
-            self.bbox.origin = tuple(o + dist for o in self.bbox.origin)
-            self.bbox.shape = (newsize, newsize)
             raise UpdateException
 
         # grow the box?
         # because the PSF moves power across the box, the gradients at the edge
         # accummulate flux from beyond the box
-        if image.m is not None:
+        elif image.m is not None:
```

Comment:
No need to use an `elif`, since the previous `if` will always raise an exception.

Reply:
nitpicking, are we?